### ISSUE TITLE
Source Tracking recognises a deleted Global Action (`QuickAction`) as `QuickActionDefinition` and fails retrieving the change

### ISSUE THREAD

**Issue by amtrack** [forcedotcom/cli#2829]
### Summary

When deleting a Global Action in a source tracked org, `SourceMember.MemberType` is incorrectly reported as `QuickActionDefinition` instead of `QuickAction`.

`sf project retrieve` does not do anything because the Metadata Registry does not know this Metadata Type.

```
$ sf project retrieve preview
Warning: Unable to find type QuickActionDefinition in registry
...
```

### Steps To Reproduce

I created a Minimum Reproduction Example using GitHub Actions: 
https://github.com/mdapi-issues/mre-incorrect-source-member-type-quickactiondefinition

### Expected result

It should retrieve the change and delete the QuickAction locally.

### Actual result

It does not delete the remotely deleted Global Action.

### System Information

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.36.8",
  "nodeVersion": "node-v20.10.0",
  "osVersion": "Darwin 23.4.0",
  "rootPath": "/Users/john.doe/.config/yarn/global/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.13 (core)",
    "@oclif/plugin-commands 3.2.2 (core)",
    "@oclif/plugin-help 6.0.20 (core)",
    "@oclif/plugin-not-found 3.1.1 (core)",
    "@oclif/plugin-plugins 5.0.5 (core)",
    "@oclif/plugin-search 1.0.20 (core)",
    "@oclif/plugin-update 4.2.3 (core)",
    "@oclif/plugin-version 2.0.16 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.7 (core)",
    "@salesforce/cli 2.36.8 (core)",
    "apex 3.1.3 (core)",
    "auth 3.5.1 (core)",
    "data 3.2.4 (core)",
    "deploy-retrieve 3.5.3 (core)",
    "info 3.1.2 (core)",
    "limits 3.2.1 (core)",
    "marketplace 1.1.0 (core)",
    "org 3.6.3 (core)",
    "packaging 2.3.0 (core)",
    "schema 3.2.0 (core)",
    "settings 2.1.2 (core)",
    "sobject 1.2.1 (core)",
    "source 3.2.2 (core)",
    "telemetry 3.1.17 (core)",
    "templates 56.1.0 (core)",
    "trust 3.5.1 (core)",
    "user 3.4.3 (core)"
  ]
}
```
### Additional information

https://github.com/mdapi-issues/mre-incorrect-source-member-type-quickactiondefinition/assets/789765/357d0b69-a8be-439a-8a6f-b2ae1209688e

**Comment by WillieRuemmele** [comment]
once again @amtrack, thanks for the great repro, and self investigation results!

**Comment by jshackell-sfdc** [comment]
This issue is addressed in version 2.44.8 (Jun 5, 2024).
